pkg/controllers/streaming/processor_controller.go

```diff
@@ -422,6 +426,55 @@ func (r *ProcessorReconciler) constructDeploymentForProcessor(processor *streami
 		return nil, err
 	}
 
+	volumes := []corev1.Volume{}
+	volumeMounts := []corev1.VolumeMount{}
+	streams := append(inputStreams, outputStreams...)
+	streamBindings := append(processor.Spec.Inputs, processor.Spec.Outputs...)
+	for i, stream := range streams {
+		if stream.Status.Binding.MetadataRef.Name != "" {
+			metadataVolumeName := fmt.Sprintf("processor-stream-%s-metadata", stream.Name)
```

Comment:
Same line, different argument:
Because it should be possible to do things like `riff streaming processor create ... --output out --output someAlias:out`, the name here should either
- **also** be based on the alias
- stay as the stream name, but be de-duped.

Reply:
ok. let's change the parameter to be the alias instead of the stream's name, and add validation that ensures alias uniqueness.

Reply:
> let's change the parameter to be the alias instead of the stream's name, and add validation that ensures alias uniqueness.

Actually, I think the second option is the only reasonable one, because it should be ok to have the same alias twice, as long as one is on the input side and the other on the output side.

Reply:
Which causes a problem in the current design because the alias is used in the path.

Is it ok per the spec to use different root paths, including per-type (streams) but even per-role (ie `inputsBindingsRootPath vs outputsBindingsRootPath`) ?

Reply:
> it should be ok to have the same alias twice, as long as one is on the input side and the other on the output side.

I don't think it's unreasonable to have the alias be unique for both inputs and outputs. We can relax this down the road if we deconflict the file path.

> Is it ok per the spec to use different root paths

There is only one directory to look at as far as the CNB Bindings spec is concerned. We could do something different, but then we're deviating from CNB Bindings.